packages/web_ui/src/util/websocket.jsx

```diff
@@ -119,6 +123,26 @@ export class Control extends libLink.Link {
 		console.log("WS", message.data.direction, message.data.content);
 	}
 
+	async onLiveSlaveAdded(id, handler) {
+		if (this.liveUpdateSlaveHandlers.size === 0) {
+			libLink.messages.setLiveSlaveSubscription.send(this, { connect: true });
+		}
+		this.liveUpdateSlaveHandlers.set(id, handler);
+	}
+
+	async offLiveSlaveAdded(id) {
+		this.liveUpdateSlaveHandlers.delete(id);
+		if (this.liveUpdateSlaveHandlers.size === 0) {
+			libLink.messages.setLiveSlaveSubscription.send(this, { connect: false });
+		}
+	}
+
+	async liveUpdateSlavesEventHandler(message) {
+		for (let [_, handler] of this.liveUpdateSlaveHandlers.entries()) {
+			handler(message);
```

Comment:
Use the `.values()`/`.keys()` methods of a map if you only need to iterate the value/key.